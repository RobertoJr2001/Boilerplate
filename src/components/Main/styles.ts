import style from 'styled-components'

export const Wrapper = style.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = style.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = style.h1`
  font-size: 2.5rem;
`

export const Description = style.h2`
font-size: 2rem;
font-wight: 400;
`

export const Illustration = style.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`