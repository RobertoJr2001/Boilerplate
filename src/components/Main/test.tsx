import { render, screen } from '@testing-library/react'

import Main from './index'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
        screen.getByRole('heading', { name: /react avanÃ§ado/i }))
          .toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot()
  });

  it('should render the styles corretly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 
      'background-color': '#06092b', 
      'color': '#fff',
      'width': '100%',
      'height': '100%',
      'padding': '3rem',
      'text-align': 'center', 
      'display': 'flex',
      'flex-direction': 'column',
      'align-items': 'center',
      'justify-content': 'center' 
  })

  });
});