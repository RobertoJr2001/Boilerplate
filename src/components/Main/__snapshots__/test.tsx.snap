// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-18tj86z-0 eyPVIf"
>
  <img
    alt="Imagem de um átomo React Avançado escrito ao lado."
    class="styles__Logo-sc-18tj86z-1 kcognk"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-18tj86z-2 buNrEy"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-sc-18tj86z-3 eUEOvo"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Desenvolvedor sentado de frente para uma tela com código."
    class="styles__Illustration-sc-18tj86z-4 gOxUtQ"
    src="/img/hero-illustration.svg"
  />
</main>
`;
